src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -827,20 +826,16 @@ private SafeCurlSlistHandle SetRequestHeaders(SafeCurlHandle handle, HttpRequest
                 contentHeaders = request.Content.Headers;
             }
 
-            string[] allHeaders = HeaderUtilities.DumpHeaders(request.Headers, contentHeaders)
-                .Split(s_headerDelimiters, StringSplitOptions.RemoveEmptyEntries);
+            // Prepare request headers
+            List<string> allHeaders = PrepareRequestHeaders(request.Headers, contentHeaders);
```

Comment:
The System.Net.Http API surface (unlike the WinRT Windows.Web.Http API surface) allows developers to add custom headers to either the content header collection or the request/response header collection (HttpRequestMessage.Headers, HttpResponseMessage.Headers).  So, it is possible that there are "duplicate" headers (i.e. similarly named header-names, which are allowed per RFC).